tests/unit/test.config.js

```diff
@@ -259,6 +270,43 @@ describe('config', () => {
       const newArgv = applyConfigToArgv({argv, configObject, defaultValues});
       assert.strictEqual(newArgv.sourceDir, '/configured/directory');
     });
+
+    it('throws an error when an option is not camel cased', () => {
+      const {argv, defaultValues} = makeArgv({
+        globalOpt: {
+          'source-dir': {
+            type: 'string',
+            demand: false,
+          },
+        },
+      });
+      const configObject = {
+        'source-dir': 'fake/value/',
+      };
+      assert.throws(() => {
+        applyConfigToArgv({argv, configObject, defaultValues});
+      }, UsageError, 'UsageError: The config option "source-dir" must be ' +
+        'specified in camel case: "sourceDir"');
+    });
+
+    it('throws an error when an option is invalid', () => {
+      const {argv, defaultValues} = makeArgv({
+        globalOpt: {
+          'source-dir': {
+            type: 'string',
+            demand: false,
+          },
+        },
+      });
+      const configFileName = 'fake/path/to/config';
+      const configObject = {
+        artifacts: 'fake/value/',
```

Comment:
Since the check for invalid options relies on the original hyphenated key, I think it would be good to make this `artifactsDir` to cover that part of the logic.